lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -204,6 +204,18 @@ protected function _getPortableTableColumnDefinition($tableColumn)
             $column->setPlatformOption('collation', $tableColumn['collation']);
         }
 
+        // MySQL 5.7 specific https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html
+        if (strpos($tableColumn['extra'], 'GENERATED') !== false) {
+            // extra's syntax in the case of a generated columns:
+            // <UPPERCASE_GENERATION_TYPE> GENERATED
+            $pieces = explode(' ', $tableColumn['extra']);
```

Comment:
You only use `$pieces[0]` here. Maybe you'd rather use `[$generationType] = explode(' ', $tableColumn['extra']);`? Also, is there a limit in the `explode()` expression here? Do you care only about the first space? If so, add `2` as a parameter to `explode()`

Reply:
Or `$generationType = strstr($tableColumn['extra'], ' ', true);`?